src/backend.rs

```diff
@@ -634,49 +644,115 @@ impl SharedBackend {
 
     /// Returns the full block for the given block identifier
     pub fn get_full_block(&self, block: impl Into<BlockId>) -> DatabaseResult<Block> {
-        tokio::task::block_in_place(|| {
+        let block_id = block.into();
+        let place = {
             let (sender, rx) = oneshot_channel();
-            let req = BackendRequest::FullBlock(block.into(), sender);
+            let req = BackendRequest::FullBlock(block_id, sender);
             self.backend.clone().try_send(req)?;
             rx.recv()?
-        })
+        };
+        let block = {
+            let mut backend = self.backend.clone();
+            let (sender, rx) = oneshot_channel();
+            tokio::task::spawn_blocking(move || {
+                let req = BackendRequest::FullBlock(block_id, sender);
+                let _ = backend.try_send(req);
+            });
+            rx.recv()?
+        };
+        match self.blocking_mod {
+            BlockingMod::BlockInPlace => tokio::task::block_in_place(|| place),
+            BlockingMod::Block => block,
+        }
     }
 
     /// Returns the transaction for the hash
     pub fn get_transaction(&self, tx: B256) -> DatabaseResult<WithOtherFields<Transaction>> {
-        tokio::task::block_in_place(|| {
+        let place = {
             let (sender, rx) = oneshot_channel();
             let req = BackendRequest::Transaction(tx, sender);
             self.backend.clone().try_send(req)?;
             rx.recv()?
-        })
+        };
+        let block = {
+            let mut backend = self.backend.clone();
+            let (sender, rx) = oneshot_channel();
+            tokio::task::spawn_blocking(move || {
+                let req = BackendRequest::Transaction(tx, sender);
+                let _ = backend.try_send(req);
+            });
+            rx.recv()?
+        };
+        match self.blocking_mod {
+            BlockingMod::BlockInPlace => tokio::task::block_in_place(|| place),
+            BlockingMod::Block => block,
+        }
```

Comment:
here we need to match on the mode and use the block_in_place only if mode is BlockInPlace otherwise just use

```
            let (sender, rx) = oneshot_channel();
            let req = BackendRequest::Transaction(tx, sender);
            self.backend.clone().try_send(req)?;
            rx.recv()?
```

this can like be done with a helper function on blockingMode that accepts a closure so we can reuse this for all other calls like

`self.block_mode.run(|| rx.recv())`